src/astro/databases/google/bigquery.py

```diff
@@ -306,6 +312,38 @@ def get_project_id(self, target_table) -> str:
         except AttributeError:
             raise ValueError(f"conn_id {target_table.conn_id} has no project id")
 
+    def load_local_file_to_bigquery(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        native_support_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ) -> None:
+        """Transfer data from local to bigquery"""
+        native_support_kwargs = native_support_kwargs or {}
+
+        write_disposition_val = {"replace": "WRITE_TRUNCATE", "append": "WRITE_APPEND"}
+        client = self.hook.get_client()
+        config = {
+            "source_format": self.NATIVE_PATHS_SUPPORTED_FILE_TYPES[
+                source_file.type.name
+            ],
+            "create_disposition": "CREATE_IF_NEEDED",
+            "write_disposition": write_disposition_val[if_exists],
+            "autodetect": True,
+        }
+        config.update(native_support_kwargs)
+        job_config = bigquery.LoadJobConfig(config)
+
+        with open(source_file.path, "rb") as file:
```

Comment:
How do we know the file is always binary?

Reply:
@dimberman  Thanks, we don't. So I have added a property in FileTypes to convey the open modes - r/rb and used it in the open() function.

Reply:
Ah yeah so I guess if it's text we need to encode it and pass it as binary.

Reply:
Yes, for us it will always be binary read mode, irrespective of the file type.